overlays/build-tools-in-build-inputs.nix

```diff
@@ -4,17 +4,112 @@ let
   inherit (import ../lib { inherit lib; }) checkMkDerivationFor;
 
   buildTools = [
+    "antlr"
+    "asciidoc"
+    "asciidoctor"
+    "autoconf"
+    "autogen"
+    "automake"
+    "automoc4"
+    "autoreconfHook"
+    "bazel"
+    "bc"
+    "bdf2psf"
+    "bison"
+    "bmake"
+    "breakpointHook"
+    "bzip2"
+    "bundler"
     "cmake"
+    "docutils"
+    "dos2unix"
+    "desktop-file-utils"
+    "docbook2x"
+    "docbook_xml_dtd_412"
+    "docbook_xml_dtd_42"
+    "docbook_xml_dtd_43"
+    "docbook_xml_dtd_44"
+    "docbook_xml_dtd_45"
+    "docbook-xsl-nons"
+    "docbook-xsl-ns"
+    "ensureNewerSourcesForZipFilesHook"
+    "extra-cmake-modules"
+    "flex"
+    "gawk"
+    "gcc"
+    "gcc10"
+    "getopt"
+    "git"
+    "gnum4"
+    "gnumake"
+    "gnused"
+    "gnustep.make"
+    "gogUnpackHook"
+    "go-md2man"
+    "gtk-doc"
+    "gzip"
+    "help2man"
+    "intltool"
+    "libtool"
+    "lit"
+    "lld"
+    "llvm"
+    "llvmPackages"
+    "makeSetupHook"
+    "makeWrapper"
+    "man"
+    "maven"
     "meson"
     "ninja"
+    "nodePackages.node-gyp"
+    "nodePackages.node-gyp-build"
+    "nodePackages.node-pre-gyp"
+    "pandoc"
+    "patch"
+    "patchelf"
     "pkg-config"
+    "poetry"
+    "pythonPackages.pytest"
+    "python3Packages.pytest"
+    "qmake4Hook"
+    "qt5.wrapQtAppsHook"
+    "ronn"
+    "rpcsvc-proto"
+    "rpm"
+    "rpmextract"
+    "rsync"
+    "pythonPackages.setuptools-git"
+    "pythonPackages.setuptools-scm"
+    "pythonPackages.setuptools-scm-git-archive"
+    "pythonPackages.sphinx"
+    "squashfsTools"
+    "sudo"
+    "subversion"
+    "swig"
+    "unzip"
+    "updateAutotoolsGnuConfigScriptsHook"
+    "util-linux"
+    "vala"
+    "wafHook"
+    "which"
+    "wrapGAppsHook"
+    "pythonPackages.wrapPython"
+    "xcbuildHook"
+    "xcodebuild"
+    "xmlto"
+    "xvfb_run"
+    "yacc"
+    "yarn"
+    "zip"
+  ] ++ lib.optionals (prev.stdenv.isDarwin or false) [
```

Comment:
There should be no need for `or false`, `isDarwin` should always be in `stdenv`.